ci/docker/runtime_functions.sh

```diff
@@ -880,6 +880,41 @@ nightly_test_javascript() {
     make -C /work/mxnet/amalgamation libmxnet_predict.js MIN=1 EMCC=/work/deps/emscripten/emcc
 }
 
+# Nightly 'MXNet: The Straight Dope' Single-GPU Tests
+set_up_nightly_straight_dope_tests() {
+    set -ex
+    cd /work/mxnet/tests/nightly/straight_dope
+    rm -rf ./straight_dope_book
+    git clone https://github.com/zackchase/mxnet-the-straight-dope straight_dope_book
```

Comment:
Would it be possible to do this as part of the test class init? That way, the test would be self contained and doesn't need any bootstrapping.